docs/source/tutorials/Parameters.ipynb

```diff
@@ -0,0 +1,247 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0041971d-0f17-4ee4-9b6a-81bea8c51369",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sim.get_graph(True, True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "298aa458-e518-424d-af9d-1ce45d55e4f4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Accessing a parameter and giving it a value will turn it into a static parameter\n",
+    "sim.SIE.phi = 0.4\n",
+    "sim.get_graph(True, True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "08e6a094-5630-45cd-ab6b-1dbd89b3ebad",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Accessing a parameter and setting it to None will turn it into a dynamic parameter\n",
+    "sim.lenslight.x0 = None\n",
+    "sim.get_graph(True, True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3f6bbccd-ae91-444f-a8ac-ec31b1562d78",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This also gives us the order of parameters for a vector that can be an input to the sim function\n",
+    "x_tens = torch.tensor(\n",
+    "    [\n",
+    "        1.5,  # z_s\n",
+    "        0.5,  # sie z_l\n",
+    "        0.1,  # sie x0\n",
+    "        -0.1,  # sie y0\n",
+    "        0.7,  # sie cosmo h0\n",
+    "        0.31,  # sie cosmo Om0\n",
+    "        0.0,  # source x0\n",
+    "        0.0,  # source y0\n",
+    "        0.7,  # source q\n",
+    "        1.4,  # source phi\n",
+    "        0.1,  # lenslight x0\n",
+    "        -0.1,  # lenslight y0\n",
+    "        0.6,  # lenslight q\n",
+    "        3.0,  # lenslight n\n",
+    "        1.0,  # lenslight Ie\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "# alternatively we can construct a dictionary\n",
+    "x_dict = {\n",
+    "    \"sim\": {\n",
+    "        \"z_s\": torch.tensor(1.5),\n",
+    "    },\n",
+    "    \"SIE\": {\n",
+    "        \"z_l\": torch.tensor(0.5),\n",
+    "        \"x0\": torch.tensor(0.1),\n",
+    "        \"y0\": torch.tensor(-0.1),\n",
+    "    },\n",
+    "    \"cosmo\": {\n",
+    "        \"h0\": torch.tensor(0.7),\n",
+    "        \"Om0\": torch.tensor(0.31),\n",
+    "    },\n",
+    "    \"source\": {\n",
+    "        \"x0\": torch.tensor(0.0),\n",
+    "        \"y0\": torch.tensor(0.0),\n",
+    "        \"q\": torch.tensor(0.7),\n",
+    "        \"phi\": torch.tensor(1.4),\n",
+    "    },\n",
+    "    \"lenslight\": {\n",
+    "        \"x0\": torch.tensor(0.1),\n",
+    "        \"y0\": torch.tensor(-0.1),\n",
+    "        \"q\": torch.tensor(0.6),\n",
+    "        \"n\": torch.tensor(3.0),\n",
+    "        \"Ie\": torch.tensor(1.0),\n",
+    "    },\n",
+    "}\n",
+    "\n",
+    "fig, axarr = plt.subplots(1, 2, figsize=(16, 8))\n",
+    "axarr[0].imshow(\n",
+    "    sim(sim.pack(x_tens), quad_level=3).detach().cpu().numpy(), origin=\"lower\"\n",
```

Comment:
Change this to `sim(x)` instead. It's better that the user uses our built-in forward method instead of the pack method. The pack method should almso be considered as a private method.